server/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineTransportAction.java

```diff
@@ -33,7 +34,7 @@
 import java.util.Map;
 import java.util.Random;
 
-public class SimulatePipelineTransportAction extends HandledTransportAction<SimulatePipelineRequest, SimulatePipelineResponse> {
+public class SimulatePipelineTransportAction extends TransportAction<SimulatePipelineRequest, SimulatePipelineResponse> {
```

Comment:
This class shows SOLUTION 2C: registration code is inside the action, but as a static create+register method.